addons/io_hubs_addon/third_party/recast.py

```diff
@@ -0,0 +1,636 @@
+        default=EDGE_MAX_LENGTH,
+        min=0.0,
+        max=30.0,
+        subtype='DISTANCE')
+
+    verts_per_poly: IntProperty(
+        name="verts_per_poly",
+        description="Verts per poly",
+        default=VERTS_PER_POLY_DEFAULT,
+        min=3,
+        max=10
+    )
+
+    sample_dist: FloatProperty(
+        name="sample_dist",
+        description="Sample distance",
+        default=SAMPLE_DIST_DEFAULT,
+        min=0.0,
+        max=30.0,
+        subtype='DISTANCE')
+
+    sample_max_error: FloatProperty(
+        name="sample_max_error",
+        description="Max sample error",
+        default=SAMPLE_MAX_ERROR_DEFAULT,
+        min=0.0,
+        max=30.0,
+        subtype='DISTANCE')
+
+    partitioning: EnumProperty(
+        name="partitioning",
+        items=[("WATERSHED", "WATERSHED", "WATERSHED"),
+               ("MONOTONE", "MONOTONE", "MONOTONE"),
+               ("LAYERS", "LAYERS", "LAYERS")],
+        default=PARTITIONING_DEFAULT)
+
+    color: FloatVectorProperty(name="Color",
+                               description="Color",
+                               subtype='COLOR_GAMMA',
+                               default=COLOR_DEFAULT,
+                               size=4,
+                               min=0,
+                               max=1)
+    expanded: BoolProperty(name="expanded", default=True)
+
+    auto_cell: BoolProperty(name="Auto Cell", default=AUTO_CELL_DEFAULT)
+
+
+class RecastNavMeshPanel(Panel):
+    """Creates a Panel in the Object properties window"""
+    bl_label = "Recast navmesh"
+    bl_idname = "SCENE_PT_blendcast"
+    bl_space_type = 'PROPERTIES'
+    bl_region_type = 'WINDOW'
+    bl_context = "scene"
+
+    def draw(self, context):
+        layout = self.layout
+        recastPropertyGroup = context.scene.recast_navmesh
+
+        layout.operator("recast.build_navigation_mesh")
+        layout.operator("recast.reset_navigation_mesh")
+        layout.prop(recastPropertyGroup, "color", text="Color")
+
+        layout.label(text="Rasterization:")
+        col = layout.column()
+        col.row().prop(recastPropertyGroup, "auto_cell", text="Auto Cell")
+        if not recastPropertyGroup.auto_cell:
+            col.row().prop(recastPropertyGroup, "cell_size", text="Cell size")
+        col.row().prop(recastPropertyGroup, "cell_height", text="Cell height")
+
+        layout.label(text="Agent:")
+        col = layout.column()
+        col.row().prop(recastPropertyGroup, "agent_height", text="Height")
+        col.row().prop(recastPropertyGroup, "agent_radius", text="Radius")
+        col.row().prop(recastPropertyGroup, "slope_max", text="Maximum slope")
+        col.row().prop(recastPropertyGroup, "climb_max", text="Maximum step height")
+
+        layout.label(text="Region:")
+        col = layout.column()
+        col.row().prop(recastPropertyGroup, "region_min_size", text="Min region size")
+        col.row().prop(recastPropertyGroup, "partitioning", text="Partitioning")
```

Comment:
I completely forgot about this in the first review, but I think the partitioning property should be included in advanced settings too.